### PR TITLE
Migrates ReputationToken to Solidity.

### PR DESCRIPTION
**Pull request by MicahZoltu**
This is an important step because it shows a contract working with the delegation stuff.  It also brings to light an issue that is going to be uncomfortable during the migration, which is the need for interface definitions for any not-yet-migrated dependencies.

### CODE REVIEW

**Comment by joeykrug** [comment]
Build failed

**Comment by joeykrug** [comment]
> which is the need for interface definitions for any not-yet-migrated dependencies.

I dont follow

**Comment by MicahZoltu** [comment]
See `ReputationToken.sol`, we have to define an interface for `Branch` at the top because Branch isn't migrated yet.

**Comment by MicahZoltu** [comment]
Ugh.  The style checker doesn't support multi-line boolean conditionals.  :'(

**Comment by MicahZoltu** [comment]
Fixing for now, but we should consider disabling that check: https://travis-ci.org/AugurProject/augur-core/builds/257026528#L581

**Comment by MicahZoltu** [comment]
https://travis-ci.org/AugurProject/augur-core/builds/257026528#L583 is also a stupid rule (leads to mistakes).  By putting the operator on the left, it makes it clear to the reader at a glance that the line is a continuation.  When you put it on the line at the right, it is easy to not notice that both lines are part of the same condition because reading non-aligned right side is much harder than aligned left side.

File: src/reporting/ReputationToken.sol [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    string public name = "Reputation";
+    string public symbol = "REP";
+    uint256 public decimals = 18;
+    Branch public branch;
+    ReputationToken public topMigrationDestination;
+
+    function initialize(Branch _branch) public returns (bool) {
+        require(_branch != address(0));
+        require(branch == address(0));
```
Comment (joeykrug) [inline src/reporting/ReputationToken.sol]: These _branch vs branch seem kinda confusing, other than that looks good

Comment (MicahZoltu) [inline src/reporting/ReputationToken.sol]: This is fixed in my latest PR.  I also found it confusing.

**Pull request merged**